Tasks/InstallSSHKeyV0/installsshkey-util.ts

```diff
@@ -241,12 +294,54 @@ export function tryRestoreKnownHosts() {
     let knownHostsDeleteFileOnExit: string = tl.getTaskVariable(postKnownHostsDeleteFileSetting);
     
     tl.debug('Restoring known_hosts');
-    if (knownHostsDeleteFileOnExit && knownHostsLocation) {
-        fs.unlinkSync(knownHostsLocation);
-    } else if (knownHostsContents && knownHostsLocation) {
-        fs.writeFileSync(knownHostsLocation, knownHostsContents);
-    } else if (knownHostsLocation || knownHostsContents) {
-        tl.warning(tl.loc('CannotResetKnownHosts'));
-        tl.debug('(location=' + knownHostsLocation + ' content=' + knownHostsContents + ')');
+    tryRestore('known_hosts', knownHostsContents, knownHostsLocation, knownHostsDeleteFileOnExit);
+}
+
+export function tryRestoreConfig() {
+    let configContents: string = tl.getTaskVariable(postConfigContentsSetting);
+    let configLocation: string = tl.getTaskVariable(postConfigLocationSetting);
+    let configDeleteFileOnExit: string = tl.getTaskVariable(postConfigDeleteFileSetting);
+    
+    tl.debug('Restoring config');
+    tryRestore('config', configContents, configLocation, configDeleteFileOnExit);
+}
+
+export function tryDeletePrivateKeyFile(privateKeyFileID: string) {
+    if (privateKeyFileID) {
+        tl.debug(tl.loc("DeletePrivateKeyFile"));
+        let secureFileHelpers: SecureFileHelpers = new SecureFileHelpers();
+        secureFileHelpers.deleteSecureFile(privateKeyFileID);
+    }
+}
+
+export class ConfigFileEntry {
+    public alias: string;
+    public port: string;
+    public hostName: string;
+    public identityFile: string;
+    public user: string;
+
+    public constructor(alias:string, hostName: string, user: string, identityFile: string, port: string) {
+        this.alias = alias;
```

Comment:
Instead of simple assiging - you can define constructor arguments with public/private keyword